test/PowerNotifications/PowerTests.cpp

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "winrt/Microsoft.ProjectReunion.h"
+
+using PowerEventHandle =
+winrt::Windows::Foundation::EventHandler<winrt::Windows::Foundation::IInspectable>;
+
+using namespace winrt::Microsoft::ProjectReunion;
+
+namespace ProjectReunionPowerTests
+{
+    constexpr auto TIMEOUT = 2000;
+    class PowerTests
+    {
+    public:
+        BEGIN_TEST_CLASS(PowerTests)
+            TEST_CLASS_PROPERTY(L"ActivationContext", L"PowerTests.dll.manifest")
```

Comment:
@DrusTheAxe DO these tests need to run for different IL and AC levels like Environmentmanager?

Reply:
Possibly

General rule of thumb, you want RunAs:RestrictedUser as that ensures the test runs as a 'normal' user (not elevated etc) regardless who invoked TE.EXE. If you have concerns about normal vs elevated invocation (or other user contexts) then you'll want to specify multiple RunAs, or multiple tests with different fixtures if need be.

I wouldn't say it's _required_ for tests to run at varying IL/AC. That's something each feature team needs to judge. But if there are differences or notable scenarios then yes, it'd be wise to do so.

Also remember, this is Unit and Functional Tests. You also have opportunity to run Integration Tests at the top of the stack using the end NuGet+MSIXs. Some tests may be better suited for one vs the other. Ideally you do as much as you can here  so problems are detected early but YMMV.